scripts/cangen/main.py

```diff
@@ -68,6 +68,14 @@ def parse():
     bus_list = config["canGen"]["busses"]
     output_path = config["canGen"].get("outputPath", DEFAULT_OUTPUT_DIR)
 
+
+#clear the directory file
+if os.path.exists(output_path):
+    for filename in os.listdir(output_path):
+        os.remove(os.path.join(output_path, filename))
```

Comment:
`os.listdir()` also returns the names of subfolders of `output_path`. this would cause `os.remove(...` to fail since it only works on files. We want to delete the entire `generated/can` folder. You could try to recursively search for directories and folders, but a simpler solution is to use [`shutil.rmtree`](https://docs.python.org/3/library/shutil.html#shutil.rmtree) which deletes a folder and its contents.

If you use this function, you can replace the `for` loop with just `shutil.rmtree(output_path)`. This also makes it clearer that we are deleting the entire directory.